### PR TITLE
Rework docstrings to the Sphinx style

### PR DESCRIPTION
**Pull request by PerchunPak**
I want to merge this into master, so it will not produce many merge conflicts in possible future.

### CODE REVIEW

File: mcstatus/address.py [hunk 1]
```diff
@@ -229,3 +232,7 @@ async def async_minecraft_srv_address_lookup(
         port = default_port
 
     return Address(host, port)
+
+
+# some hacks for the documentation
+del _AddressBase.__new__.__doc__
```
Comment (ItsDrike) [inline mcstatus/address.py:238]: Why is this necessary? Can't such an ignore be done directly in `docs/`? I'd rather not clutter the actual code with it.

File: mcstatus/protocol/connection.py [hunk 2]
```diff
@@ -409,25 +398,24 @@ async def read_buffer(self) -> Connection:
         return result
 
 
-class BaseConnection:
+class BaseConnection:  # todo: maybe use abstract class?
```
Comment (ItsDrike) [inline mcstatus/protocol/connection.py:401]:
```suggestion
class BaseConnection:
```

This isn't related to docstrings, if you want to see this, make a github issue instead. I don't like the idea of keeping tons of TODO comments directly in code, let's use the bug tracker on github directly, where we can discuss whether this makes sense, who will implement it, and how it could be implemented.

File: mcstatus/utils.py [hunk 3]
```diff
@@ -131,6 +132,7 @@ def deprecated(
     msg: str | None = None,
     methods: Iterable[str] | None = None,
 ) -> Callable | type[object]:
+    # TODO add ``.. deprecated::`` to docstring
```
Comment (ItsDrike) [inline mcstatus/utils.py:135]: Same remarks as with my other comment regarding TODOs in code.

**Review by ItsDrike — CHANGES_REQUESTED** [review]
I'd like to see an extension like `flake8-docstrings` added too, so that there's something enforcing the proper docstring format is followed.

**Comment by PerchunPak** [comment]
> I'd like to see an extension like `flake8-docstrings` added too, so that there's something enforcing the proper docstring format is followed.

It will be in the next PR, this just fixes issues like using one ` instead of two.

File: mcstatus/protocol/connection.py (hunk 2 above)
Comment (PerchunPak) [reply mcstatus/protocol/connection.py]: Created #459, removed [here](https://github.com/py-mine/mcstatus/compare/b4c7e9763c5d6f7138586c81b52947c1894b9073..bb194c20f1882cd4832076a1360070ec9bffce64).

File: mcstatus/utils.py (hunk 3 above)
Comment (PerchunPak) [reply mcstatus/utils.py]: Created #460, removed [here](https://github.com/py-mine/mcstatus/compare/b4c7e9763c5d6f7138586c81b52947c1894b9073..bb194c20f1882cd4832076a1360070ec9bffce64).

File: mcstatus/address.py (hunk 1 above)
Comment (PerchunPak) [reply mcstatus/address.py]: Removed [here](https://github.com/py-mine/mcstatus/compare/bb194c20f1882cd4832076a1360070ec9bffce64..405901f3971b240e4b3b576c409cadbfe0b121cf) and added to `conf.py` [here](https://github.com/py-mine/mcstatus/compare/7ac401423b7425a434d8f428cde745e6745533cd..cfc4b3e5fa9254fd6e6cd7294f7be796a729d04d).

**Review by ItsDrike — APPROVED** [review]
Thanks!

**Review by ItsDrike — COMMENTED** [review]
Would you mind also making sure that the docstrings in `tests/` are consistent with this new style? While these won't be used in the docs, we shouldn't be using 2 different docstring conventions in the codebase. So if there are some, please adapt them too.

**Comment by PerchunPak** [comment]
> Would you mind also making sure that the docstrings in `tests/` are consistent with this new style? While these won't be used in the docs, we shouldn't be using 2 different docstring conventions in the codebase. So if there are some, please adapt them too.

Done [here](https://github.com/py-mine/mcstatus/compare/405901f3971b240e4b3b576c409cadbfe0b121cf..b3c53aea555d0d4c734fe91d81441d925a1bfafd)!
